### PR TITLE
Refund bets, fix blinds timer, allow ff to showdown if blinded all-in

### PR DESCRIPTION
**Pull request by teddylo57**
-refund bets after showdown
-fix blinds timer
-allow fast forward to showdown if you are blinded all-in

### CODE REVIEW

File: backend/src/main/java/poker/models/game/Table.java [hunk 1]
```diff
@@ -140,6 +141,28 @@ public void setButton(int seatNumber) {
         .collect(Collectors.toList());
   }
 
+  List<Seat> getSeatsRemainingToActAndAllIn() {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/game/Table.java]: nice, descriptive name

File: backend/src/main/java/poker/models/game/Table.java [hunk 2]
```diff
@@ -140,6 +141,28 @@ public void setButton(int seatNumber) {
...
+    return getSeatsInHand().stream()
+        .filter(seat -> seat.getPlayer().isYetToAct())
+        .filter(seat -> seat.getPlayer().isAllIn())
+        .collect(Collectors.toList());
+  }
+
+  public boolean allInFastForwardBlinds() {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/game/Table.java]: something like `doBlindsPutPlayersAllIn`? not entirely sure what `allInFastForwardBlinds` means

File: backend/src/main/java/poker/models/game/Table.java [hunk 3]
```diff
@@ -140,6 +141,28 @@ public void setButton(int seatNumber) {
...
+    if (getSeatsRemainingToActAndAllIn().size() > 0) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/game/Table.java]: would another way to approach this issue to set player `isAllIn` to `true` when setting the blinds?

then we could re-use `shouldFastForwardToShowdown`?

File: backend/src/main/java/poker/models/game/Table.java [hunk 4]
```diff
@@ -247,6 +271,18 @@ public void evaluateWinner(List<Player> players) {
     players.stream()
         .map(Player::getUsername)
         .forEach(s -> Logger.getLogger(Table.class).info("[WINNERS] " + s));
+
+    List<Player> sortedPlayers =
+    getSeatsInHand().stream().map(Seat::getPlayer).sorted(new TotalBetAmountComparator()).collect(Collectors.toList());
+
+    if (sortedPlayers.size() >= 2 && (sortedPlayers.get(0).getTotalHandBetAmount()) > sortedPlayers.get(1).getTotalHandBetAmount()) {
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/game/Table.java]: 1 step closer to side pots

File: backend/src/main/java/poker/models/game/Table.java [hunk 5]
```diff
@@ -298,6 +334,7 @@ public void reset() {
               player.setYetToAct(false);
               player.setAllIn(false);
             });
+    this.increaseBlinds = false;
```
Comment (ericliu121187) [inline backend/src/main/java/poker/models/game/Table.java]: how come we are removing this?

File: backend/src/main/java/poker/models/game/Table.java (hunk 3 above)
Comment (teddylo57) [reply backend/src/main/java/poker/models/game/Table.java]: Hm, yeah but if the player that is all-in is the big blind, we still want the other player to act, and to call the big blind.

If the player that is all-in is the small blind, we want to fast forward

File: backend/src/main/java/poker/models/game/Table.java (hunk 5 above)
Comment (teddylo57) [reply backend/src/main/java/poker/models/game/Table.java]: Adding it in, to reset the table to the initial state

File: backend/src/main/java/poker/models/game/Table.java (hunk 3 above)
Comment (ericliu121187) [reply backend/src/main/java/poker/models/game/Table.java]: got it -- from a product perspective, i can't think of a situation where a player who is all-in can still act

Comment (teddylo57) [reply backend/src/main/java/poker/models/game/Table.java]: i'll add that in the refactor, cleaning up the code

**Pull request merged**